e2e/test/Helpers/logging/MsTestLogger.cs

```diff
@@ -1,20 +1,33 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Runtime.Versioning;
+using Microsoft.ApplicationInsights.DataContracts;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 
 namespace Microsoft.Azure.Devices.E2ETests
 {
     /// <summary>
     /// Ms Test framework specific logging.
+    /// Every test will have its own instance of this logger with properties specific to the test.
+    /// This logger uses the global logger to write logs but has its own set of properties.
+    /// We cannot inherit and log using this because this logger object gets disposed at the end of the test
+    /// and we will lose logs without doing a flush. Flush is expensive and should be done only once when
+    /// we exit the test suite.
     /// </summary>
-    public class MsTestLogger : TestLogger
+    public class MsTestLogger
```

Comment:
I did not understand the bit about the problem with having this class inherit from `TestLogger`; could you explain that part a bit more?